src/neuroglancer/sliceview/volume/renderlayer.ts

```diff
@@ -0,0 +1,449 @@
+    builder.addVarying('highp vec3', 'vChunkPosition');
+
+    builder.setVertexMain(`
+vec3 chunkSize = uChunkDataSize * uVoxelSize;
+int vertexIndex = int(aVertexIndexFloat);
+for (int e = 0; e < 4; ++e) {
+  highp ivec2 vidx = uVertexIndex[vertexIndex*4 + e];
+  highp vec3 v1 = max(uLowerClipBound, min(uUpperClipBound, chunkSize * uVertexBasePosition[vidx.x] + uTranslation));
+  highp vec3 v2 = max(uLowerClipBound, min(uUpperClipBound, chunkSize * uVertexBasePosition[vidx.y] + uTranslation));
+  highp vec3 vDir = v2 - v1;
+  highp float denom = dot(vDir, uPlaneNormal);
+  if (abs(denom) > ${ORTHOGONAL_EPSILON}) {
+    highp float lambda = (uPlaneDistance - dot(v1, uPlaneNormal)) / denom;
+    if ((lambda >= -${LAMBDA_EPSILON}) && (lambda <= (1.0 + ${LAMBDA_EPSILON}))) {
+      lambda = clamp(lambda, 0.0, 1.0);
+      highp vec3 position = v1 + lambda * vDir;
+      gl_Position = uProjectionMatrix * vec4(position, 1.0);
+      vChunkPosition = (position - uTranslation) / uVoxelSize;
+      break;
+    }
+  }
+}
+`);
+
+    builder.addFragmentCode(glsl_getPositionWithinChunk);
+  }
+
+  computeVerticesDebug(
+      uChunkDataSize: vec3, uVoxelSize: vec3, uPlaneDistance: number, uPlaneNormal: vec3,
+      uTranslation: vec3, uProjectionMatrix: mat4) {
+    let chunkSize = vec3.multiply(vec3.create(), uChunkDataSize, uVoxelSize);
+    let frontVertexIndex = findFrontVertexIndex(uPlaneNormal);
+    let uVertexIndex =
+        this.data.vertexIndices.subarray(frontVertexIndex * 48, (frontVertexIndex + 1) * 48);
+    let vidx = [0, 0];
+    let v = [vec3.create(), vec3.create()];
+    let vStart = vec3.create(), vDir = vec3.create(), position = vec3.create(),
+        gl_Position = vec3.create(), vChunkPosition = vec3.create();
+    let vertexBasePositions = new Float32Array(this.data.vertexBasePositions);
+    let uVertexBasePosition = (i: number) => <vec3>vertexBasePositions.subarray(i * 3, i * 3 + 3);
+    for (let vertexIndex = 0; vertexIndex < 6; ++vertexIndex) {
+      for (let e = 0; e < 4; ++e) {
+        for (let j = 0; j < 2; ++j) {
+          vidx[j] = uVertexIndex[2 * (vertexIndex * 4 + e) + j];
+          vec3.multiply(v[j], chunkSize, uVertexBasePosition(vidx[j]));
+        }
+        vec3.add(vStart, v[0], uTranslation);
+        vec3.subtract(vDir, v[1], v[0]);
+        let denom = vec3.dot(vDir, uPlaneNormal);
+        if (Math.abs(denom) > ORTHOGONAL_EPSILON) {
+          let lambda = (uPlaneDistance - vec3.dot(vStart, uPlaneNormal)) / denom;
+          if ((lambda >= -LAMBDA_EPSILON) && (lambda <= 1.0 + LAMBDA_EPSILON)) {
+            lambda = Math.max(0, Math.min(1, lambda));
+            vec3.scaleAndAdd(position, vStart, vDir, lambda);
+            vec3.transformMat4(gl_Position, position, uProjectionMatrix);
+            vec3.scale(vChunkPosition, uVertexBasePosition(vidx[0]), 1.0 - lambda);
+            vec3.scaleAndAdd(vChunkPosition, vChunkPosition, uVertexBasePosition(vidx[1]), lambda);
+            console.log(
```

Comment:
Perhaps this is just the work of clang-format, but better formatting would be nice.

(This function also hasn't worked for a little while, since it hasn't been updated to reflect changes in other code, but it is still somewhat useful to keep around so that it could be fixed later if necessary for debugging.)